src/core/services/interimElement/interimElementBase.js

```diff
@@ -0,0 +1,217 @@
+      notifyShowing(options.scope, element, options, controller);
+
+      var defer = $q.defer();
+
+      try {
+        $q.when(options.onShow(options.scope, element, options, controller))
+          .then(function() {
+
+            notifyComplete(options.scope, element, options);
+            startAutoHide();
+
+            defer.resolve(element);
+
+          }, defer.reject)
+
+      } catch (err) {
+        defer.reject(err.message);
+      }
+
+      return defer.promise;
+    }
+
+    function hideElement() {
+      var defer = $$q.defer();
+      var announceRemoving = options.onRemoving || angular.noop;
+
+      try {
+        var action = $$q.when(options.onRemove(options.scope, element, options) || true);
+
+        announceRemoving(element, action);
+
+        if (options.$destroy) {
+          defer.resolve(element);
+        } else {
+
+          action.then(function() {
+
+            if (!options.preserveScope && options.scope) {
+              options.scope.$destroy();
+            }
+
+            defer.resolve(element);
+
+          }, defer.reject);
+
+        }
+
+      } catch (err) {
+        defer.reject(err.message);
+      }
+
+      return defer.promise;
+    }
+
+    function prepareInterimElement() {
+      if (options.template) {
+        options.template = $mdUtil.processTemplate(options.template);
+      }
+
+      // Apply default options to the interim element.
+      options = angular.extend({
+        preserveScope: false,
+        cancelAutoHide : angular.noop,
+        scope: options.scope || $rootScope.$new(options.isolateScope),
+
+        onShow: function transitionIn(scope, element, options) {
+          return $animate.enter(element, options.parent);
+        },
+
+        onRemove: function transitionOut(scope, element) {
+          return element && $animate.leave(element) || $q.when();
+        }
+      }, options);
+    }
+
+
+    function compileElement(options) {
+      return options.skipCompile ? mockCompile() : $mdCompiler.compile(options);
+
+      function mockCompile() {
+        return $q.when({
+          locals: {},
+          link: function () {
+            return options.element;
+          }
+        });
+      }
+    }
+
+    function linkElement(compiledData) {
+      angular.extend(compiledData.locals, options);
+
+      element = compiledData.link(options.scope);
+
+      options.element = element;
+      options.parent = findParent(element, options);
+
+      if (options.themable) $mdTheming(element);
+
+      return element;
+
+    }
+
+    function findParent(element, options) {
+      var parent = options.parent;
+
+      if (angular.isFunction(parent)) {
+        parent = parent(options.scope, element, options);
+      } else if (angular.isString(parent)) {
+        parent = angular.element($document[0].querySelector(parent));
+      } else {
+        parent = angular.element(parent);
+      }
+
+      if (!(parent || {}).length) {
```

Comment:
Yep, had the same thoughts before, but then no parent wasn't found. So I changed it back.